tests/compute/nonuniformres-as-function-parameter.slang

```diff
@@ -31,65 +37,74 @@ MyStruct func(RWStructuredBuffer<uint> buffer)
 
-    // CHECK-SPV: OpDecorate %[[VAR1:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-SPV: OpDecorate %[[VAR2:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_SPV: OpDecorate %[[VAR1:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_SPV: OpDecorate %[[VAR2:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_SPV-NOT: OpDecorate %[[VAR3:[a-zA-Z0-9_]+]] NonUniform
 
 
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR1:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR2:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR3:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR4:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR5:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR6:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR7:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR8:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR9:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR10:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR11:[a-zA-Z0-9_]+]] NonUniform
-    // CHECK-GLSL-SPV: OpDecorate %[[VAR12:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR1:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR2:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR3:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR4:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR5:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR6:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR7:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR8:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR9:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR10:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR11:[a-zA-Z0-9_]+]] NonUniform
+    // CHECK_GLSL_SPV: OpDecorate %[[VAR12:[a-zA-Z0-9_]+]] NonUniform
 
-    // CHECK-GLSL: layout(local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
 
+    // CHECK_GLSL_SPV: OpCopyObject %int %{{.*}}
+    uint bufferIdx = pixelIndex.x;
     uint nonUniformIdx = NonUniformResourceIndex(bufferIdx);
     RWStructuredBuffer<uint> buffer = globalBuffer[nonUniformIdx];
 
-    // CHECK-SPV: %[[VAR3:[a-zA-Z0-9_]+]] = OpBitcast %int %bufferIdx
-    // CHECK-SPV: %[[VAR1]] = OpAccessChain %_ptr_StorageBuffer_RWStructuredBuffer{{.*}} %{{.*}} %[[VAR3]]
+    // CHECK_SPV: %[[VAR1]] = OpAccessChain %_ptr_StorageBuffer_RWStructuredBuffer{{.*}} %{{.*}} %bufferIdx
 
-    // CHECK-GLSL-SPV: %[[VAR1:[a-zA-Z0-9_]+]] = OpCopyObject %int %{{.*}}
-    // CHECK-GLSL-SPV: %[[VAR2:[a-zA-Z0-9_]+]] = OpAccessChain %_ptr_Uniform_uint %globalBuffer_0 %[[VAR1]] %int_0 %int_0
-    // CHECK-GLSL-SPV: %[[VAR3:[a-zA-Z0-9_]+]] = OpLoad %uint %[[VAR2]]
+    // CHECK_GLSL_SPV: %[[VAR1:[a-zA-Z0-9_]+]] = OpCopyObject %int %{{.*}}
```

Comment:
should this be just `[[VAR1]]` since `VAR1` is already captured in the above `OpDecorate` lines?

Reply:
Right, I will just remove [[VAR1]] here. it's a typo.

Reply:
Oh, after looking at it carefully, it's not a typo, it's on purpose.
Because we need to check the inst assigning to VAR is decorated as `NonUniform`

Reply:
you can see VAR1 ~ VAR12 are all decorated as `NonUniform`.

Reply:
ACK